packages/gatsby-image/src/index.js

```diff
@@ -16,26 +16,32 @@ const convertProps = props => {
   return convertedProps
 }
 
+// Find the source of an image to use as a key in the image cache.
+// Use `fixed` or `fluid` if specified, or the first image in
+// either `fixedImages` or `fluidImages`
+const getImageSrc = props =>
```

Comment:
Not a massive fan of this. This might be cleaner if we overloaded `fluid` and `fixed` to take an Array, instead of introducing new props.